addons/mail/models/mail_thread.py

```diff
@@ -4507,7 +4508,31 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
             # sudo: mail.message.translation - discarding translations of message after editing it
             self.env["mail.message.translation"].sudo().search([("message_id", "=", message.id)]).unlink()
             payload["Message"]["translationValue"] = False
-        self.env["bus.bus"]._sendone(message._bus_notification_target(), "mail.record/insert", payload)
+        notifications = [[message._bus_notification_target(), "mail.record/insert", payload]]
+        if outdated_starred_partners:
+            # sudo: bus.bus: reading non-sensitive last id
+            bus_last_id = self.env["bus.bus"].sudo()._bus_last_id()
+            notifications.extend([
+                [
+                    partner,
+                    "mail.record/insert",
+                    {
+                        "Thread": {
+                            "id": "starred",
+                            "messages": [("DELETE", {"id": empty_messages.ids})],
+                            "model": "mail.box",
+                            "counter": count,
+                            "counter_bus_id": bus_last_id,
+                        }
+                    },
+                ]
+                for partner, count in self.env["mail.message"]._read_group(
```

Comment:
I don't agree with the reasoning here, it only does the extra query when deleting a message, which is a not common operation, and the query is not a complex query on the message table, it's a stupid simple query based on the pivot table which is fully indexed, with as many partners to return as there are starring a message (which should be close to 0 and therefore not even done in most cases).

Don't do premature optimization, and if the ORM is making a stupid query, we can always do it manually.

We're wasting more and more time constantly handling feedback or tickets about this, on top of race conditions on runbot that are annoying for everyone due to bugged and inconsistent code.

If we want to use discuss some day, it needs to be reliable and to be working well, otherwise people will keep having a bad opinion of it. This kind of issue is part of the problem, and it needs to be fixed.

Reply:
Discord frequently has issues with counters, and it is not an issue. Synchronizing details like that are not an issue. Some people might complain but it is like bikeshedding: if we don't use discuss internally that's not because multitabs synchronization when voiding a message is not working top-notch.

Here it is not about ORM optimization, it is about useless code. We have to write it, test it, maintain it, upgrade it, just for a flow that is becoming slower for a niche use case. That's typically where we have to say "ok, that's not a big issue, we prefer simpler code".

I agree there is a gray zone, but globally feedbacks that matter are the one making the product move forward, not loosing time coding fixes in stable that will globally never bring new people using discuss.

Reply:
This issue in particular is wasting everyone's time since 2 years (introduction of the delete feature).

We regularly have the feedback, we have testers reporting the issue when testing tasks with these flows, we have an open task in our pipe for this specific bug, we have new developers asking why at some point, we have race conditions in tests, we are now wasting time discussing about it...

I keep in mind to not do complex code for the sake of it in general, but for this case I'd much prefer maintaining the extra code rather than keeping this issue open forever.

Reply:
> This issue in particular is wasting everyone's time since 2 years (introduction of the delete feature).

Yeah, I feel like you might be exaggerating a bit. Note that the whole delete feature implementation was flagged as being source of issues since the beginning, so maybe it should have been thought a bit more precisely at the beginning. We indeed saw a lot of ugly code spawning a bit everywhere because of that feature.